internals/overlord/overlord.go

```diff
@@ -427,20 +424,14 @@ func Fake() *Overlord {
 func FakeWithState(handleRestart func(restart.RestartType)) *Overlord {
 	o := &Overlord{
 		loopTomb: new(tomb.Tomb),
-		inited:   false,
 	}
 	s := state.New(fakeBackend{o: o})
 	o.stateEng = NewStateEngine(s)
 	o.runner = state.NewTaskRunner(s)
 	return o
 }
 
-// AddManager adds a manager to a fake overlord. It cannot be used for
-// a normally initialized overlord those are already fully populated.
 func (o *Overlord) AddManager(mgr StateManager) {
-	if o.inited {
-		panic("internal error: cannot add managers to a fully initialized Overlord")
-	}
```

Comment:
We agreed to use the ```o.StateEngine().AddManager(...)``` directly, instead of modifying the ```o.AddManager(...)``` test method (for use with FakeOverlord), for adding managers. This allows us to leave the ```inited``` flag and its usage unchanged for now.